packages/discord.js/src/structures/UncachedGuildMember.js

```diff
@@ -0,0 +1,332 @@
+'use strict';
+
+const Base = require('./Base');
+const TextBasedChannel = require('./interfaces/TextBasedChannel');
+const { GuildMemberFlagsBitField } = require('../util/GuildMemberFlagsBitField');
+const PermissionsBitField = require('../util/PermissionsBitField');
+
+/**
+ * Represents a member of a guild on Discord. Used in interactions from guilds that aren't cached.
+ * Backwards compatible with {@link APIGuildMember}.
+ * @implements {TextBasedChannel}
+ * @extends {Base}
+ */
+class UncachedGuildMember extends Base {
+  constructor(client, data, guildId) {
+    super(client);
+
+    /**
+     * The guild that this member is part of
+     * @type {Guild}
+     */
+    this.guildId = guildId;
+
+    if ('user' in data) {
+      /**
+       * The user that this guild member instance represents
+       * @type {?User}
+       */
+      this.user = this.client.users._add(data.user, true);
+    }
+
+    /**
+     * The nickname of this member, if they have one
+     * @type {?string}
+     */
+    this.nickname = data.nick;
+
+    /**
+     * The guild member's avatar hash
+     * @type {?string}
+     */
+    this.avatar = data.avatar ?? null;
+
+    /**
+     * The role ids of the member
+     * @name UncachedGuildMember#roleIds
+     * @type {Snowflake[]}
+     */
+    this.roleIds = data.roles;
+
+    /**
+     * The timestamp the member joined the guild at
+     * @type {number}
+     */
+    this.joinedTimestamp = Date.parse(data.joined_at);
+
+    /**
+     * The timestamp the member joined the guild at, as an ISO8601 timestamp
+     * @type {string}
+     * @deprecated Use {@link UncachedGuildMember#joinedTimestamp} or {@link UncachedGuildMember#joinedAt} instead
+     */
+    this.joined_at = data.joined_at;
+
+    /**
+     * The last timestamp this member started boosting the guild
+     * @type {?number}
+     */
+    this.premiumSinceTimestamp = data.premium_since ? Date.parse(data.premium_since) : null;
+
+    /**
+     * The last timestamp this member started boosting the guild, as an ISO8601 timestamp
+     * @type {?string}
+     * @deprecated Use {@link UncachedGuildMember#premiumSinceTimestamp}
+     * or {@link UncachedGuildMember#premiumSince} instead
```

Comment:
```suggestion
     * @deprecated Use {@link UncachedGuildMember#premiumSinceTimestamp} or {@link UncachedGuildMember#premiumSince} instead
```

Reply:
Richardkouakouchangeascommit.12fr

Reply:
this is what I had originally, but it fails prettier due to the line length, so I split it up.